application/models/Files.php

```diff
@@ -139,9 +139,26 @@ public static function addFile(Model_Users $user, Array $fileData, $thumbnail_da
 			return self::PROCESS_OK;
 		}
 		
-		public static function addFolder(Model_Users $user, $name = null, $description = null, Model_Files $parent = null)
+		public static function addFolder(Model_Users $user, $name = null, $description = null, Model_Files $parent = null, $tags = null)
 		{
-			$folder	=	new Model_Files($name, $description, 1, $user, $parent);
+			
+			//We manage the tags
+			$arrayTags	=	explode(' ', $tags);
+			$arrayTagsInstances = [];
+			
+			foreach($arrayTags as $tagName)
+			{
+				$tag = Model_Tags::getTag($tagName);
```

Comment:
Tu es sûr que c'est avec un NOT IN qu'il faut faire ? Parce que ça va seulement aller chercher tous les tags dans la BDD qu'on a pas lié à notre document. Et c'est pas ce qu'on veut.
Je pense plutôt faire une requête avec un IN, puis faire une boucle avec in_array() de PHP. On aura toujours la boucle, mais plus les requêtes SQL dedans d'accord.